### PR TITLE
 Refactor charges controller

### LINKED ISSUE #40: Refactor charges/cart controllers to make them skinnier

### PR DESCRIPTION
**Pull request by noobjey**
closes #40 
- pulled stripe payments into payment processor
- model tests for payment processor
- refactored charge controller

### CODE REVIEW

File: app/models/payment_processor.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class PaymentProcessor
+  attr_accessor :customer,
+                :card_token,
+                :charge
+
+  def initialize(email, card_token, amount)
+    @email       = email
```
Comment (imwithsam) [inline app/models/payment_processor.rb:7]: extra space

File: spec/models/payment_processor_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  it "creates a stripe customer" do
+
+    pp.make_payment
+
+    expect(pp.customer.email).to eq(user_email)
+    expect(pp.customer.card).to eq(fake_token)
+  end
+
+  it "creates charge for the customer" do
+
```
Comment (imwithsam) [inline spec/models/payment_processor_spec.rb:31]: extra blank line

File: spec/models/payment_processor_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    Stripe.api_key = 'pk_test_Sc54Tr3cr3BehQcnGzyCz5xu'
+    StripeMock.start
+  end
+
+  after do
+    StripeMock.stop
+  end
+
+  it "creates a stripe customer" do
+
```
Comment (imwithsam) [inline spec/models/payment_processor_spec.rb:23]: extra blank line

File: spec/models/payment_processor_spec.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  it "returns true if payment successfull" do
+    expect(pp.make_payment).to eq(true)
+  end
+
+  it "returns false if payment not successful" do
+    StripeMock.prepare_card_error(:card_declined)
+
+    expect(pp.make_payment).to eq(false)
+  end
+
```
Comment (imwithsam) [inline spec/models/payment_processor_spec.rb]: extra blank line

File: app/models/payment_processor.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  def create_stripe_customer
+    Stripe::Customer.create(
+      email: @email,
+      card:  card_token
+    )
+  end
+
+
```
Comment (imwithsam) [inline app/models/payment_processor.rb]: extra blank lines

**Comment by dsshim** [comment]
Nice work on stripe!

**Pull request merged**